pkg/issuer/helper.go

```diff
@@ -65,7 +73,14 @@ func (h *helperImpl) GetGenericIssuer(ref cmmeta.ObjectReference, ns string) (cm
 		if h.clusterIssuerLister == nil {
 			return nil, fmt.Errorf("cannot get ClusterIssuer named %q as cert-manager is scoped to a single namespace", ref.Name)
 		}
-		return h.clusterIssuerLister.Get(ref.Name)
+		o, err := h.clusterIssuerLister.Get(ref.Name)
+		if err != nil {
+			return o, err
+		}
+		o = o.DeepCopy()
+		// Apply defaults in case a defaulting webhook has not been deployed
+		webhook.Scheme.Default(o)
```

Comment:
It feels really awkward that we have to inject these reference to calling `Default` everywhere, and I feel like in some parts of our codebase we will perform a Get or a List operation and *not* call `Default`, meaning that our functions need to continue to assume that objects are not defaulted (else we'll introduce subtle and hard to spot bugs).

I'd be more inclined to suggest we 1) begin defaulting in the webhook if it is installed and 2) continue to assume that defaults are not applied in our code, and have explicit handling code (like we do now) that handles the `if not set` logic.

Whilst it does mean we are duplicating that defaulting logic, it'll mean it's always clear what is going on in our code and doesn't create a fracture between methods that expect objects to be pre-defaulted and those that do not.

Reply:
I've moved the scheme to `pkg/internal/api` but I can't import it from there into the `cmd/webhook/app/start.go` because it's `pkg/internal` rather than `./internal`.
So I've left an alias in place at `pkg/webhook`